### PR TITLE
daemon: return "snapname_rev.snap" style when using /v2/download

### PR DESCRIPTION
**Pull request by mvo5**
This PR changs the /v2/download endpoint to return a more useful
filename for the downloaded snap. It uses the form
"snapName_rev.snap", e.g. "core_1234.snap". The old code was
just using the snap name which is a bit too limiting.

This will help with the `snap download` via the snapd API code.

### CODE REVIEW

File: daemon/response.go [hunk 1]
```diff
@@ -251,15 +251,15 @@ func makeErrorResponder(status int) errorResponder {
 // A FileStream ServeHTTP method streams the snap
 type fileStream struct {
 	SnapName string
-	Info     snap.DownloadInfo
+	Info     *snap.Info
```
Comment (pedronis) [inline daemon/response.go]: it would make more sense to take an explicit filename and the DownloadInfo than hard coding messaging info in response.go

Comment (mvo5) [reply daemon/response.go]: Thank you! I fixed this now.

File: daemon/api_download.go [hunk 2]
```diff
@@ -85,7 +86,8 @@ func streamOneSnap(c *Command, user *auth.UserState, snapName string) Response {
 
 	return fileStream{
 		SnapName: snapName,
-		Info:     info,
+		FileName: filepath.Base(info.MountFile()),
```
Comment (pedronis) [inline daemon/api_download.go]: nitpick: we are not super consistent, but Filename is by far the more common spelling in the codebase

**Comment by mvo5** [comment]
@pedronis Thanks, I fixed the nitpick now. There is just a single test update needed in https://github.com/snapcore/snapd/pull/7515/files#diff-f79c87c123f7d405c029820fd2920898R197

**Comment by pedronis** [comment]
@mvo, ok, seems I need more tea :)

**Pull request merged**